web/src/components/apps/Dashboard.jsx

```diff
@@ -232,6 +234,38 @@ class Dashboard extends Component {
     });
   }
 
+  fetchAppDownstream = async () => {
+    const { app } = this.props;
+    if (!app) return;
+
+    try {
+      const res = await fetch(`${window.env.API_ENDPOINT}/app/${app.slug}`, {
+        headers: {
+          "Authorization": Utilities.getToken(),
+          "Content-Type": "application/json",
+        },
+        method: "GET",
+      });
+      if (res.ok && res.status == 200) {
+        const app = await res.json();
+        if (app?.downstreams?.length > 0 && app?.downstreams[0].pendingVersions?.length > 0) {
+          const isAwaitingResults = app.downstreams[0].pendingVersions[0].status === "pending_preflight" || app.downstreams[0].pendingVersions[0].status === "deploying" || app.downstreams[0].pendingVersions[0].status === "unknown";
```

Comment:
if this logic exists in the version history too, we should move it to a shared function

Reply:
@sgalsaleh updated this.